File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDiffInBST(self, root: Optional[TreeNode]) -> int:
        #flatten tree into list
        
        
        def traverse(point, lst):
            
            if point:
                traverse(point.left, lst)
                lst.append(point.val)
                traverse(point.right, lst)
            return lst

        lst = sorted(traverse(root, []))
        
        #find diff
        
        
        return min([lst[i] - lst[i-1] for i in range(1, len(lst))])

        

        
        

File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
import java.util.*;
class Solution {
    public ArrayList<Integer> traverse(TreeNode a, ArrayList<Integer> lst){
        if (a != null){
            this.traverse(a.left, lst);
            lst.add(a.val);
            this.traverse(a.right, lst);
        }
        return lst;
    }

    public Integer smallestDiff(ArrayList<Integer> lst){
        Integer prev = lst.get(0);
        ArrayList<Integer> res = new ArrayList<>();
        for (Integer i = 1; i< lst.size(); i++){
            Integer temp = lst.get(i) - prev;
            prev = lst.get(i);
            res.add(temp);
        }
        Collections.sort(res);
        return res.get(0);
    }
    
    public int minDiffInBST(TreeNode root) {
        
        ArrayList<Integer> lst = new ArrayList<>();
        lst = traverse(root, lst);
        Integer diffn = smallestDiff(lst);

        return (int) diffn;
    }
}